cider/tests/utils/CiderRunnerTest.cpp

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include <gtest/gtest.h>
+#include <string>
+
+#include "CiderRunner.h"
+#include "QueryDataGenerator.h"
+
+std::string create_ddl() {
+  return R"(CREATE TABLE test(col_1 INTEGER, col_2 BIGINT,
+  col_3 FLOAT, col_4 DOUBLE);)";
+}
+
+std::shared_ptr<CiderBatch> create_default_batch() {
+  return std::make_shared<CiderBatch>(
+      QueryDataGenerator::generateBatchByTypes(1000,
+                                               {"col_1", "col_2", "col_3", "col_4"},
+                                               {CREATE_SUBSTRAIT_TYPE(I32),
+                                                CREATE_SUBSTRAIT_TYPE(I64),
+                                                CREATE_SUBSTRAIT_TYPE(Fp32),
+                                                CREATE_SUBSTRAIT_TYPE(Fp64)}));
+}
+
+TEST(CiderRunnerTest, statelessRunnerMultiBatchTest) {
```

Comment:
Are these tests aimed at that cider will not crash?

Reply:
yes